opentimelineio/adapters/cmx_3600.py

```diff
@@ -126,17 +126,26 @@ def add_clip(self, line, comments, rate=24):
                 edl_rate
             )
 
-            duration = record_out - record_in
-            if duration != clip_handler.clip.duration():
-                if self.ignore_timecode_mismatch:
-                    # use the source
-                    record_out = record_in + duration
+            src_duration = clip_handler.clip.duration()
+            rec_duration = record_out - record_in
+            if rec_duration != src_duration:
+                motion = comment_handler.handled.get('motion_effect')
+                freeze = comment_handler.handled.get('freeze_frame')
+                if motion is not None or freeze is not None:
+                    # Adjust the clip to match the record duration
+                    # TODO: Once OTIO has speed effects, use them here.
+                    clip_handler.clip.source_range.duration = rec_duration
+
+                elif self.ignore_timecode_mismatch:
+                    # Pretend there was no problem by adjusting the record tc
+                    record_out = record_in + src_duration
```

Comment:
Still a little weirded out that `record_out` is never used again, but I kinda understand why.